examples/arrowir/test_arrowir.cpp

```diff
@@ -0,0 +1,589 @@
+	}
+
+	case duckdb::LogicalOperatorType::LOGICAL_FILTER: {
+		auto &filter = (duckdb::LogicalFilter &)op;
+		if (filter.expressions.size() == 0) {
+			// this is a bug in the plan generation but we can work around this
+			return transform_op(fbb, *op.children[0]);
+		}
+		// pretty ugly this
+		auto filter_expr_arrow = transform_expr(fbb, *filter.expressions[0]);
+		// loop from second element ^^
+		auto function_name = fbb.CreateString("and");
+		for (auto i = begin(filter.expressions) + 1, e = end(filter.expressions); i != e; ++i) {
+			auto rhs_arrow = transform_expr(fbb, *i->get());
+			auto arg_list = fbb.CreateVector<flatbuffers::Offset<arrowir::Expression>>({filter_expr_arrow, rhs_arrow});
+			auto call = arrowir::CreateCall(fbb, function_name, arg_list);
+			filter_expr_arrow =
+			    arrowir::CreateExpression(fbb, arrowir::ExpressionImpl::ExpressionImpl_Call, call.Union());
+		}
+
+		auto arrow_filter = arrowir::CreateFilter(fbb, rel_base, transform_op(fbb, *op.children[0]), filter_expr_arrow);
+		return arrowir::CreateRelation(fbb, arrowir::RelationImpl_Filter, arrow_filter.Union());
+	}
+
+	case duckdb::LogicalOperatorType::LOGICAL_COMPARISON_JOIN: {
+		auto &join = (duckdb::LogicalComparisonJoin &)op;
+		arrowir::JoinKind arrow_join_kind;
+		switch (join.join_type) {
+		case duckdb::JoinType::INNER:
+			arrow_join_kind = arrowir::JoinKind_Inner;
+			break;
+		case duckdb::JoinType::SEMI:
+			arrow_join_kind = arrowir::JoinKind_LeftSemi;
+			break;
+		case duckdb::JoinType::RIGHT:
+			arrow_join_kind = arrowir::JoinKind_RightOuter;
+			break;
+		case duckdb::JoinType::ANTI:
+			arrow_join_kind = arrowir::JoinKind_Anti;
+			break;
+		default:
+			throw std::runtime_error("unsupported join type");
+		}
+
+		auto arrow_join_expression = transform_join_condition(fbb, join.conditions[0]);
+		auto function_name = fbb.CreateString("and");
+		for (auto i = begin(join.conditions) + 1, e = end(join.conditions); i != e; ++i) {
+			auto rhs_arrow = transform_join_condition(fbb, *i);
+			auto arg_list =
+			    fbb.CreateVector<flatbuffers::Offset<arrowir::Expression>>({arrow_join_expression, rhs_arrow});
+			auto call = arrowir::CreateCall(fbb, function_name, arg_list);
+			arrow_join_expression =
+			    arrowir::CreateExpression(fbb, arrowir::ExpressionImpl::ExpressionImpl_Call, call.Union());
+		}
+
+		auto arrow_join =
+		    arrowir::CreateJoin(fbb, rel_base, transform_op(fbb, *op.children[0]), transform_op(fbb, *op.children[1]),
+		                        arrow_join_expression, arrow_join_kind);
+		return arrowir::CreateRelation(fbb, arrowir::RelationImpl_Join, arrow_join.Union());
+	}
+
+	default:
+		throw std::runtime_error(duckdb::LogicalOperatorToString(op.type));
+	}
+}
+
+static duckdb::LogicalType transform_type_back(const arrow::Field *field) {
+	switch (field->type_type()) {
+	case arrow::Type_Binary:
+		return duckdb::LogicalType::VARCHAR;
+	case arrow::Type_Int: // TODO which size is this type? unclear.
```

Comment:
I think you need to use the `type_as_Foo` methods in cases where the type has some associated data. For `Int` I think you'd do something like this:

```c++
case arrow::Type_Int:
  switch (field->type_as_Int()->bitWidth) {
    case 1:
      return TINYINT;
    case 2:
      return INT;
    // etc
  }
```